marshal.go

```diff
@@ -144,3 +145,157 @@ func (i InstanceMetadata) MarshalXML(e *xml.Encoder, start xml.StartElement) err
 	// flush to ensure tokens are written
 	return e.Flush()
 }
+
+// startLocalName creates a start-tag of an XML element with the given local name and no namespace name.
+func startLocalName(local string) xml.StartElement {
+	return xml.StartElement{Name: xml.Name{Space: "", Local: local}}
+}
+
+type metadataMap map[string]string
+
+// MarshalXML is a custom XML marshaler for metadataMap, mapping each metadata name/value pair to a
+// correspondingly named XML element with the pair's value as character data content.
+func (m metadataMap) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
+	if err := e.EncodeToken(start); err != nil {
+		return err
+	}
+
+	for k, v := range m {
+		if err := e.EncodeElement(v, startLocalName(k)); err != nil {
+			return err
+		}
+	}
+
+	return e.EncodeToken(start.End())
+}
+
+// UnmarshalXML is a custom XML unmarshaler for metadataMap, mapping each XML element's name and
+// character data content to a corresponding metadata name/value pair.
+func (m metadataMap) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
+	var v string
+	for {
+		t, err := d.Token()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		if k, ok := t.(xml.StartElement); ok {
+			if err := d.DecodeElement(&v, &k); err != nil {
+				return err
+			}
+			m[k.Name.Local] = v
+		}
+	}
+	return nil
+}
+
+func metadataValue(i DataCenterInfo) interface{} {
+	if i.Name == Amazon {
+		return i.Metadata
+	}
+	return metadataMap(i.AlternateMetadata)
```

Comment:
My use of "alternate" here is apparently a North American meaning, which is nearly synonymous with "alternative". I did some reading, and found [this explanation](https://jakubmarian.com/alternate-vs-alternative-the-difference/) (_"Alternate' in American English_):

> In American English, the word alternate is commonly used in the sense of “alternative” when the primary option is no longer available, as in
> >The road is closed. We will have to find an alternate route.

> Some speakers of American English even consider the use of “alternative” to be incorrect in this case, claiming “alternative” implies that the original option is still available.
>
> On the other hand, most Britons consider the usage of “alternate” in this sense to be a mistake and would say “alternative route” instead.

[Another source](http://grammarist.com/usage/alternate-alternative/) has this to say:
> An *alternate* is something or someone that serves in place of another. An *alternative* is a second option that does not replace the first. For example, when a road undergoing maintenance is closed to traffic, you have to take an alternate route. But when an under-construction road is still accessible to traffic, you might choose to take an alternative route to avoid congestion. The first option is still there, and the alternative gives you a choice.

Here, our first choice is to use Amazon-related metadata. Failing that, we fall back to using the _alternate_ metadata. To my ear, "alternative" makes it sound like we could choose either of them, but they should never both be available or used at the same time.

It sounds like my choice might confuse our readers in UK, but their preferred form sounds strange to my North American ear. What to do?